embedded/store/ongoing_tx.go

```diff
@@ -182,6 +184,27 @@ func (tx *OngoingTx) Set(key []byte, md *KVMetadata, value []byte) error {
 	return nil
 }
 
+func (tx *OngoingTx) AddKVConstraint(c *KVConstraints) error {
+	if tx.closed {
+		return ErrAlreadyClosed
+	}
+
+	if c == nil {
+		return ErrIllegalArguments
+	}
+
+	if len(c.Key) == 0 {
+		return ErrNullKey
+	}
+
+	if len(c.Key) > tx.st.maxKeyLen {
```

Comment:
this is showing up that the validation of the constraint datatype is not be self-contained and probably it's better for readability to have input parameters as (key, constraint)...